src/OperatorMsg.chpl

```diff
@@ -290,27 +370,65 @@ module OperatorMsg
           when (DType.Bool, DType.Int64) {
             var l = toSymEntry(left,bool);
             var val = try! value:int;
+            if boolOps.contains(op) {
+              var e = st.addEntry(rname, l.size, bool);
+              return doBinOpvs(l, val, e, op, dtype, rname, pn, st);
+            }
             var e = st.addEntry(rname, l.size, int);
             return doBinOpvs(l, val, e, op, dtype, rname, pn, st);
           }
           when (DType.Int64, DType.Bool) {
             var l = toSymEntry(left,int);
             var val = try! value.toLower():bool;
+            if boolOps.contains(op) {
+              var e = st.addEntry(rname, l.size, bool);
+              return doBinOpvs(l, val, e, op, dtype, rname, pn, st);
+            }
             var e = st.addEntry(rname, l.size, int);
             return doBinOpvs(l, val, e, op, dtype, rname, pn, st);
           }
           when (DType.Bool, DType.Float64) {
             var l = toSymEntry(left,bool);
             var val = try! value:real;
+            // Can't add boolOps.contains because chpl doesn't support bool < float
+            if boolOps.contains(op) {
+              var e = st.addEntry(rname, l.size, bool);
+              return doBinOpvs(l, val, e, op, dtype, rname, pn, st);
+            }
```

Comment:
I am a little confused by the comment here and the code. To me it looks like you are adding the piece that the comment says you can't. Am I missing something?

Reply:
you're totally right! That comment was from before Ben suggested a way to make it work, I'll pull that out